aonvif/client.py

```diff
@@ -274,9 +284,34 @@ def __init__(
 
         # Currently initialized services
         self._services = {}
+        self._capabilities = capabilities
+        self._validate_capabilities()
 
         self.to_dict = ONVIFService.to_dict
 
+    def _validate_capabilities(self):
+        if self._capabilities is None:
```

Comment:
Validation after attribute assignment looks weird. I think validation should go before assignment.

Reply:
Fixed